src/main/java/com/blazemeter/taurus/junit/reporting/TaurusReporter.java

```diff
@@ -0,0 +1,152 @@
+package com.blazemeter.taurus.junit.reporting;
+
+import com.blazemeter.taurus.junit.Utils;
+import org.json.JSONObject;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class TaurusReporter {
+
+    private FileWriter outStream;
+    private static final Logger log = Logger.getLogger(TaurusReporter.class.getName());
+    private final LinkedBlockingQueue<Sample> queue = new LinkedBlockingQueue<>();
+
+    private int activeThreads = 0;
+
+    private volatile boolean isStopped = false;
+    private final SampleFormatter formatter;
+    private final Thread reporter;
+
+    private final boolean isVerbose;
+
+    public TaurusReporter(String fileName) {
+        File file = new File(fileName);
+        try {
+            outStream = new FileWriter(file);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to open file " + fileName, e);
+        }
+
+        formatter = createFormatter(file);
+        isVerbose = formatter instanceof JSONFormatter;
+        log.info("File: " + fileName + ", formatter: " + formatter.getClass().getSimpleName());
+
+        reporter = new PoolWorker();
+        reporter.setName("Reporter thread");
+        reporter.setDaemon(true);
+        reporter.start();
+    }
+
+    private SampleFormatter createFormatter(File file) {
+        if (Utils.getFileExtension(file).equals("csv")) {
+            CSVFormatter formatter = new CSVFormatter();
+            try {
+                outStream.write(formatter.getHeader());
+                outStream.write("\n");
+            } catch (IOException e) {
+                log.log(Level.WARNING, "Failed to write CSV header", e);
```

Comment:
You can't just continue without this header written. It has to exit thread. ANd supervisor has to look at reporter's `isAlive()` and exit if it has died.